fontir/src/glyph.rs

```diff
@@ -402,6 +402,15 @@ impl Work<Context, WorkId, WorkError> for GlyphOrderWork {
             }
         }
 
+        if !context.flags.contains(Flags::ALLOW_COMPONENT_TRANSFORM) {
```

Comment:
I think we want to apply this optional decomposing operation at the beginning, before the default mixed-glyphs decomposition (in ufo2ft this is usually set as a "pre" filter, one that precedes the default filters), and also before the flatten-components operation. It should not change the results, but I think you end up doing less work in the following component-only operations if you have already removed some of the components.

Reply:
actually nevermind this last comment, doing it after flatten-components is ok and in fact matches the order in which gftools builder requests these filters to be applied:

https://github.com/googlefonts/gftools/blob/b074d8fbb52631030e6c1f524513a5a2eb7b2d56/Lib/gftools/builder/__init__.py#L358-L371
https://github.com/googlefonts/gftools/blob/b074d8fbb52631030e6c1f524513a5a2eb7b2d56/Lib/gftools/builder/_ninja.py#L203-L206